components/function-controller/internal/controllers/knative/service_controller_test.go

```diff
@@ -146,10 +153,17 @@ var _ = ginkgo.Describe("KService controller", func() {
 		reconciler = NewServiceReconciler(resourceClient, log.Log, config)
 		request = ctrl.Request{NamespacedName: types.NamespacedName{Namespace: namespace, Name: srvName}}
 	})
+	ginkgo.AfterEach(func() {
+		sel := labels.SelectorFromSet(map[string]string{testCleanupLabelKey: testCleanupLabelValue})
+		err := resourceClient.DeleteAllBySelector(ctx, &servingv1.Revision{}, namespace, sel)
```

Comment:
Revision are not deleted automatically after removing Service?

Reply:
Not when using test apiserver, we do not have knative controller, do not have k8s controller which handles owner references etc